### PR TITLE
Add Schema.org

### LINKED ISSUE #2693: Annotate `qiskit.org` with `schema.org` markup
### Feature description

In order to provide better transparency and search optimization to our pages it would be great to annotate pages of `qiskit.org` with `schema.org` markup tools. 

There are hundreds of types of annotations, which will enable better user experience when searching for information about Qiskit. 

Look at example of schema for Target corporation to get a sense what it will provide from visual perspective (not talking about better search optimization).

what-is-schema-markup-rich-results-google-SERP

Vue component for schema already exists. It need to be reviewed and integrated if acceptable. 
https://github.com/vueuse/schema-org

### Additional context

Schema.org is a collaborative, community activity with a mission to create, maintain, and promote schemas for structured data on the Internet, on web pages, in email messages, and beyond.

Schema.org is a reference website that publishes documentation and guidelines for using [structured data](https://en.wikipedia.org/wiki/Structured_data) mark-up on web-pages (called microdata). Its main objective is to standardize HTML tags to be used by webmasters for creating rich results (displayed as visual data or infographic tables on search engine results) about a certain topic of interest

### PR DESCRIPTION
**Pull request by memolina2323**
## Changes
 Added Schema.org in index.vue file with some basic information about Qiskit.org, improving SEO.
This closes #2693 

## Implementation details

Had to import nuxt's schema-org-vue.
Added some of the basic Identities, but more can be added. More info on the available ones can be found [here](https://unhead-schema-org.harlanzw.com/guide/guides/identity)

### CODE REVIEW

**Review by techtolentino — CHANGES_REQUESTED** [review]
@memolina2323 - thanks for this PR

As discussed in the vertical thread, please separate the `.gitignore` / `members.json` items, into its own pull request — this will give us a clearer understanding of the history, should we ever need to refer back to it.

Thank you

File: pages/index.vue [hunk 1]
```diff
@@ -29,4 +35,15 @@ const { data: packageInfo } = await useAsyncData<PackageInfo>(() =>
 );
 
 const qiskitVersion = packageInfo.value?.info.version ?? "";
+
+useSchemaOrg([
```
Comment (techtolentino) [inline pages/index.vue:39]: If I'm not mistaken, this should go in the [default.vue layout](https://github.com/Qiskit/qiskit.org/blob/schema-espe/layouts/default.vue) ([according to this sentence from the nuxt-schema-org docs](https://unhead-schema-org.harlanzw.com/integrations/nuxt/module#:~:text=This%20should%20be,Component%20API)).

Something to consider: 
Should we also be adding this to the `default-max.vue`, and the `metal.vue` layouts as well?

**Review by techtolentino — CHANGES_REQUESTED** [review]

File: pages/index.vue (hunk 1 above)
Comment (memolina2323) [reply pages/index.vue:39]: That is the recommended behavior, but as we're using default.vue as the header default, and schema.org usually goes inside the body tag,  for me it made more sense to include it there. 
Also, it only needs to be added in one place in the whole website.

File: pages/index.vue [hunk 2]
```diff
@@ -29,4 +35,15 @@ const { data: packageInfo } = await useAsyncData<PackageInfo>(() =>
...
+  defineWebSite({
+    name: "Qiskit.org",
+  }),
+  defineOrganization({
+    name: "Qiskit",
+    logo: "public/images/qiskit-logo.png",
```
Comment (techtolentino) [inline pages/index.vue:45]: @memolina2323 - can you please double-check if we need the `public/` part of this file path? 

It seems like with this, it'll render to:

`https://qiskit-org-pr-3243.dcq4xc5i083.us-south.codeengine.appdomain.cloud/public/images/qiskit-logo.png`

I used [this Schema Markup Validator](https://validator.schema.org/#url=https%3A%2F%2Fqiskit-org-pr-3243.dcq4xc5i083.us-south.codeengine.appdomain.cloud%2F), to test the Branch preview URL, but let me know if I'm doing that incorrectly.

Thanks!

File: pages/index.vue (hunk 1 above)
Comment (techtolentino) [reply pages/index.vue:39]: Thanks for the explanation

File: pages/index.vue (hunk 2 above)
Comment (memolina2323) [reply pages/index.vue:45]: I think you're right, the public part is not needed, thanks for pointing that out!

**Review by techtolentino — DISMISSED** [review]
Thank you for this PR @memolina2323, and for addressing my requests. I think this is good.

Again, we won't fully know if everything's working until we merge and see what the search engines decide to do w/ the schema info. [Here's the place where I read about it being included on the Home or About page.](https://www.semrush.com/blog/schema-markup-for-company-corporations/#:~:text=It%20is%20vital%20to%20add%20schema%20structured%20markup%20to%20your%20company%20%E2%80%9CAbout%E2%80%9D%20or%20Home%20page)

Let's see if @y4izus agrees.

File: nuxt.config.ts [hunk 3]
```diff
@@ -28,6 +32,7 @@ export default defineNuxtConfig({
         ? "ffdYLviQze3kzomaINXNk6NwpY9LlXcw"
         : "zbHWEXPUfXm0K6C7HbegwB5ewDEC8o1H",
       isAnalyticsEnabled: true,
+      siteUrl: process.env.NUXT_PUBLIC_SITE_URL || "https://qiskit.org",
```
Comment (y4izus) [inline nuxt.config.ts:35]: As far as I know, we don't have a environment variable called `NUXT_PUBLIC_SITE_URL`, is there something I'm missing?

File: pages/index.vue [hunk 4]
```diff
@@ -8,6 +8,12 @@
 </template>
 
 <script setup lang="ts">
+import {
+  useSchemaOrg,
+  defineWebSite,
+  defineWebPage,
+  defineOrganization,
+} from "@unhead/schema-org-vue";
```
Comment (y4izus) [inline pages/index.vue:16]: Could you double check if you need these imports?

File: pages/index.vue (hunk 1 above)
Comment (y4izus) [reply pages/index.vue:39]: I think Randy is right. By adding the `useSchemaOrg` on this page, we will only have the schema on this page. For having it on all the pages, we should add it or in the default layout or on the `app.vue` file (but we don't have it on our configuration).

You can easily check that it's not working as expected by looking at the `Elements` tab on the dev tools and looking for a line like this:

```
<script type="application/ld+json" id="schema-org-graph" ...>...</script>
```

You will notice that it's only on the main page and not on the rest

**Review by y4izus — CHANGES_REQUESTED** [review]
Good start!! let's improve and clean it a little bit ;)

File: nuxt.config.ts (hunk 3 above)
Comment (memolina2323) [reply nuxt.config.ts:35]: I was just following [this guide](https://unhead-schema-org.harlanzw.com/integrations/nuxt/module#2-configure-the-module) and it had it as a recommended guideline, but there's no further reason

File: pages/index.vue (hunk 4 above)
Comment (memolina2323) [reply pages/index.vue:16]: ooops, I added it before doing npm ci and it's not needed anymore, thanks for noticing!

File: pages/index.vue (hunk 1 above)
Comment (memolina2323) [reply pages/index.vue:39]: per offline conversations, it has been decided to be kept on index so it shows only on the main page. 👍

File: nuxt.config.ts (hunk 3 above)
Comment (memolina2323) [reply nuxt.config.ts:35]: ok, i see what you mean, let me correct that.

**Pull request merged**